x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/edit_analytics_flyout.tsx

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiOverlayMask,
+  EuiSelect,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { useMlKibana } from '../../../../../contexts/kibana';
+import { ml } from '../../../../../services/ml_api_service';
+import {
+  memoryInputValidator,
+  MemoryInputValidatorResult,
+} from '../../../../../../../common/util/validators';
+import { DataFrameAnalyticsListRow, DATA_FRAME_TASK_STATE } from './common';
+import {
+  useRefreshAnalyticsList,
+  UpdateDataFrameAnalyticsConfig,
+} from '../../../../common/analytics';
+
+interface EditAnalyticsJobFlyoutProps {
+  closeFlyout: () => void;
+  item: DataFrameAnalyticsListRow;
+}
+
+let mmLValidator: (value: any) => MemoryInputValidatorResult;
+
+export const EditAnalyticsFlyout: FC<EditAnalyticsJobFlyoutProps> = ({ closeFlyout, item }) => {
+  const [allowLazyStart, setAllowLazyStart] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
```

Comment:
Shouldn't these three properties all be initialized with the values from the config? For example, the description text box only has the existing text as the `placeholder` value. It should be populated with the current description.

Also `allow_lazy_start` should be set to `false` here.

![image](https://user-images.githubusercontent.com/7405507/86384513-bf56d580-bc86-11ea-9462-7dd786e7177b.png)

Reply:
Right now the functionality is that the placeholder is set as the current value and then we only send the field to the update endpoint if they change it. The form can't be submitted unless at least one of those has a value.

So if you only make a change to the description, what gets sent to the endpoint is just { description: new-description }

Reply:
Yes I think it is preferable to have the current values set as actual values. This makes it much easier for example to edit a simple typo in the description, and the user can also see what the current value of `allow_lazy_start` is. Is it possible to compare the edited values to the current settings to work out what has changed?

Reply:
@peteharverson - updated in 4d5f2381b5ce35656a7d8128496f92368e02ed1a 🙏